packages/flutter_tools/lib/src/runner/flutter_command.dart

```diff
@@ -1370,6 +1381,27 @@ abstract class FlutterCommand extends Command<void> {
     return dartDefineConfigJsonMap;
   }
 
+  String convertEnvFileToJsonRaw(String path, String configRaw) {
```

Comment:
This doesn't support comments, right? In my experience, most .env parsers support comments using the `#` character (which is one of the advantages of .env files over JSON. Please correct me if this is not the case.

We should either 1) support comments or 2) make it easy for the user to figure out that we don't support them (we could always add comment support if this turns out to be a popular request).

Reply:
Yes you're right. I just saw it. I'll add comment management. It's a quick win

Reply:
Comment are now managed @andrewkolos 

By default we ignore them

Feel free to give any feedback

Reply:
This doesn't support comments at the end of a line. This wouldn't necessarily be a blocker for me; however, there are cases where we could easily parse the key or value incorrectly.

1)  Comment at the end of a line:
  
      ```.env
      key=value # Here is a comment at the end of the line.
      ```
      
      Which might end up getting parsed like this:
      
      ```dart
      final entry = _parseLine('key=value # Here is the comment at the end of a line.');
      
      print(entry.value); // 'value # Here is the comment at the end of a line.'
      ```
2) Whitespace in around `=`:

      ```.env
      key = value
      ```

     Which might get parsed like this:
    
      ```dart
      final entry = _parseLine('key = value');
      
      print(entry.key); // 'key '
      print(entry.value); // ' value'
      ```

In my opinion, we need to either 1) to support comments at the end of the line (while still supporting `#` inside of a value) or 2) detect non-end-of-line comments and throw a `ToolExit` saying that we only support comments on their own line.

**I think option 2 is perfectly reasonable to do (and would be easier to implement).** However, if you decide to try option 1, then you will also need a way to allow users to still use `#` as part of an entry's value. I believe the standard way to do this is to allow users to provide values within double-quotes, single quotes, or back-ticks, like so:

```.env
key="some-#-hash-string-value"
```

This wouldn't be easy. For example, one way to parse this would be using a regex pattern such as `(\w+)(?:\s*=\s*?)\s*('(?:|[^'])*'|\s*"(?:"|[^"])*"|\s*`(?:[^`])*`|[^#\s]+)?\s*(?:#.*)?`. I would consider writing stateful parsing code first and see if the ends up being simpler than a regex.


In addition, I recommend also adding cases like these to the tests (or adding new tests for them):

```.env
# Play around with spaces around the equals sign.
key =value
key = value
key= value

key = value # Comment at the end of line. Either exclude this from the value or call throwToolExit.

# If we end up supporting comments at the end of a line:
key='value'
key="value"
key=`value`
key="'quotes inside of quotes'"
key=`"quotes`
# ... etc.
```

Reply:
On second thought, the second option of throwing a ToolExit might not be much easier, since we would still need to differentiate a `#` that is part of a value and a `#` that is meant to start a comment. There would also still be the problem of us not supporting spaces in values.

Reply:
In my opinion we need to consider env file specifications.

I'll look at this one: 

https://hexdocs.pm/dotenvy/dotenv-file-format.html

Thanks for the feedback @andrewkolos  I ping you as soon as I submit a new revision.

Reply:
Sounds good. I also think we should support the entire spec[^1], or at least detect use of features that we do not support and throw a `ToolExit` at runtime. Otherwise, we risk surprising the user with incorrectly parsed keys/values.

[^1]: To my knowledge, there actually is no official "spec", but there do seem to be docs from various libraries we can use to figure out the common set of features that are typically supported. For example, we need to detect the use of comments, multiline values, quoted values, etc. and either support them or at least throw if we cannot support them. Otherwise, we risk users  You linked one doc which seems like a good resource. I would also check out a couple of others (such as https://github.com/motdotla/dotenv/blob/master/README.md#what-rules-does-the-parsing-engine-follow) and see if there are any other features we need to consider.

Reply:
I used regex to isolate groups in the strings. 
I added all of your example (let me know if I forgot one).

One case left to manage is the multi line block
```
key="""
multi
line
block
"""
```

I already pushed the other cases waiting for this one. Feel free to challenge it.

Reply:
blocks are now managed but I faced a strange issue.

I can currently read multiline block from `.env` file content but __the result will be on a single line__ which is kinda ok I guess ?

When I tried to put some `\n` char for line separation, unit tests were successful but using this code in a sample app resulted in a strange error message saying

```bash
➜  counter flutter run -d chrome --dart-define-from-file=.env
Building flutter tool...
Resolving dependencies... (1.6s)
Got dependencies.
Launching lib/main.dart on Chrome in debug mode...
-DWELCOME_MESSAGE=Welcome%20to%20.env%20demo%0Aa%20simple%20counter%20app%20with%20.env%20file%20support%0Afor%20more%20info,%20check%20
out%20the%20README.md%20file%0AThanks!: Error: Error when reading
'-DWELCOME_MESSAGE=Welcome%20to%20.env%20demo%0Aa%20simple%20counter%20app%20with%20.env%20file%20support%0Afor%20more%20info,%20check%2
0out%20the%20README.md%20file%0AThanks!': No such file or directory
```

1. We can keep the current pushed behavior (no new line char) => no error, everything works 
2. We can try to make it work with new line char digging what's happening in the `flutter run` command. 

If we choose first option I think we are done 🎉

What is you opinion @andrewkolos ?

Reply:
I would like us to support multiline strings properly.

Unfortunately, I think we are running into a limitation of the `Process` API (from dart.io), which is what we are using to start the `dart` process later in the code. It might not support command arguments that are multiline. See this new issue, which I just filed: https://github.com/flutter/flutter/issues/129608. This has more details.

Unless you can figure out a clever way to make multiline values work, I think don't think we support them properly.
I am not sure we want to replace newlines with spaces, which I believe is what this PR currently does. It's possible that this behavior could surprise users, who would reasonably expect the newlines to be preserved.

However, I think it is reasonable to detect multiline values and call `throwToolExit` with a message explaining that multiline values are not supported.

Reply:
Thanks for the analysis about new line characters. 
I choose the `throwToolExit` option if a multi line is detected.

I also changed the tests in order to check this non managed use case.

Thank you also for creating an issue about multiline command arguments.

Tell me if the new version is Ok for you. I'm not sure about the error message. Feel free to make any suggestion.

This new code has also been tester with en brand new counter app. All works fine on my side.

Reply:
I mark this conversation as resolved in order to avoid conversation thread multiplication. Feel free to open a new one of comment the Pull Request directly